File: backend/app/api/v1/health.py
from fastapi import APIRouter
from pydantic import BaseModel
from sqlalchemy import text

from app.core.config import settings
from app.db.session import engine

router = APIRouter(tags=["health"])


class HealthResponse(BaseModel):
    status: str
    app: str
    version: str
    env: str
    database: str
    prototype_notice: str


@router.get("/health", response_model=HealthResponse)
def health() -> HealthResponse:
    try:
        with engine.connect() as conn:
            conn.execute(text("SELECT 1"))
        database = "up"
    except Exception:
        database = "down"

    return HealthResponse(
        status="ok",
        app=settings.APP_NAME,
        version=settings.VERSION,
        env=settings.APP_ENV,
        database=database,
        prototype_notice="Prototype — not an official UIDAI product.",
    )

File: frontend/src/lib/health.ts
import { z } from 'zod'
import { api } from './api'

export const healthSchema = z.object({
  status: z.string(),
  app: z.string(),
  version: z.string(),
  env: z.string(),
  database: z.string(),
  prototype_notice: z.string(),
})

export type Health = z.infer<typeof healthSchema>

export async function fetchHealth(): Promise<Health> {
  const { data } = await api.get('/health')
  return healthSchema.parse(data)
}
